### DISCUSSION TITLE
How to add Custom Species list?

### DISCUSSION THREAD

**Discussion by Vollpflock** [mcguirepr89/BirdNET-Pi#982] [Q&A]
Quick dumb question: How can I add a custom species list? I need the following: https://github.com/kahst/BirdNET-Lite/issues/4#issuecomment-995834655 (thanks to anyone who contributed)

**Comment by waynecarnold** [comment]
Under tools you can add to the custom list but note that if you put anything here, it will only look for those birds. Adding a large list would be time consuming but i guess you can add through the web terminal directly to the file.

**Reply by Vollpflock** [reply]
Yes that is my question. How can I add the list through the file manager/web terminal? It won't work if I replace or delete the existing labels.txt in web terminal.

**Reply by waynecarnold** [reply]
I would expect it to work. Ensure the proper permissions are set on the file and restart the system. I have not done this but i would think it should work. Make sure the new file is properly formatted, etc,etc.  otherwise will have to see what others say.

**Comment by Vollpflock** [comment]
Unfortunately, I can't get it to work. Anyone?

**Comment by keaneflynn** [comment]
In case anyone else finds this forum like me when looking for a way to generate a custom bird species list for your area, I made a crude repository that will generate a year-round (I know get's rid of the temporal benefit of the normal method) text file for bird species of your area. I mostly just reused the existing code, but you don't need to clone the entire birdnet repo for it to work and it gets rid of the annoying tflite dependency that seems to only work on raspberry pis with python 3.9:
[Repo](https://github.com/keaneflynn/birdnet-custom-species#)
